tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoAggregatorMapping.java

```diff
@@ -103,15 +95,11 @@ protected String getPackaging() {
     }
 
     @Override
-    protected void initModel(Model model, Reader artifactReader, File artifactFile)
-            throws ModelParseException, IOException {
+    protected void initModel(Model model, Reader artifactReader, File artifactFile) throws IOException {
         logger.debug("Generate aggregator pom for " + artifactFile);
         try (BufferedReader reader = new BufferedReader(artifactReader)) {
-            String line;
-            while ((line = reader.readLine()) != null) {
-                if (line.startsWith("#") || line.trim().isEmpty()) {
-                    continue;
-                }
+            Stream<String> lines = reader.lines().filter(l -> !l.startsWith("#") && !l.isBlank()).map(String::strip);
+            for (String line : (Iterable<String>) lines::iterator) {
```

Comment:
Why do we need a cast here? if its because of the lamda, why not calling lines.iterator()?

Reply:
Yes because of the lambda and you can only use an enhanced-for-loop with an Iterable, not with an iterator.
And without the cast, the 'assignment-target' of lines::iterator is not considered as an functional interface and therefore results in a compile error.

Reply:
I would suggest simply use while with iterator then.

Reply:
A while loop would be the longest way to use that:
```
            Iterator<String> iterator = lines.iterator();
            while (iterator.hasNext()) {
                String string = iterator.next();
```
Using a for loop with an iterator at least only has two lines:
```
            for (Iterator<String> iterator = lines.iterator(); iterator.hasNext();) {
                String line = iterator.next();
```
But the current approach has only one. But if you prefer to stick to the classic way, I suggest to use the for-loop-with-iterator.

Reply:
> But the current approach has only one. But if you prefer to stick to the classic way, I suggest to use the for-loop-with-iterator.

Having less lines is at least not a goal here and if we can save creating a lambda + obscure cast we should do that.

Reply:
> Having less lines is at least not a goal here and if we can save creating a lambda + obscure cast we should do that. You can even inline the lines variable to save a line and then we are equal in line count ;-)

That's right, but if you have less text to read (lines are IMHO not the ideal measure, since one can cheat and make it just a long line) one can also understand the code better. But I also understand that the cast+plus lambda is a unusual construct and can confuse people. It would be nice if the RHS of an enhanced for loop would behave like an assignment to a `Iterable<E>` variable. Then the case could have been omitted.

Anyway, I think the regular for loop with iterator is a good way to go. :)